kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/commands/optaplanner/AddProblemFactChangesCommand.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.server.api.commands.optaplanner;
+
+import java.util.List;
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlAttribute;
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlRootElement;
+
+import com.thoughtworks.xstream.annotations.XStreamAlias;
+import org.kie.server.api.model.KieServerCommand;
+
+@XmlRootElement(name = "add-problem-fact-changes")
+@XStreamAlias("add-problem-fact-changes")
+@XmlAccessorType(XmlAccessType.NONE)
+public class AddProblemFactChangesCommand
+        implements KieServerCommand {
+
+    private static final long serialVersionUID = -1803374525440238478L;
+
+    @XmlAttribute(name = "container-id")
+    @XStreamAlias("container-id")
+    private String containerId;
+
+    @XmlAttribute(name = "solver-id")
+    @XStreamAlias("solver-id")
+    private String solverId;
+
+    @XmlElement
+    @XStreamAlias("problem-fact-changes")
+    private List problemFactChanges;
```

Comment:
Can we make the type of this field `List<ProblemFactChange>` instead of `List`?

We might even want to use `List<? extends ProblemFactChange>`? Not sure if that adheres to Josh Bloch's PECS principle (Producer Extends, Consumer Supers) - I always get a headache there.